finder-app/writer.c

```diff
@@ -0,0 +1,83 @@
+/*@author: Divyesh  Patel
+ *@filename: writer.c
+ *@brief: ASsignment 2: File Operations and Cross Compiler
+*/
+
+#include<stdio.h>
+#include<syslog.h>
+#include<sys/types.h>
+#include<sys/stat.h>
+#include<fcntl.h>
+#include<errno.h>
+#include<unistd.h>
+#include<string.h>
+
+
+int main(int argc, char* argv[])
+{
+	//opens a connection to system logger and uses LOG_USER facility
+	openlog(NULL,0,LOG_USER);
+
+	//check is 3 arguments are provided or not
+	if(argc!=3)
+	{
+		 syslog(LOG_ERR,"Invalid Number of arguments: %d",argc); 
+		 syslog(LOG_ERR,"1.File Path"); 
+		 syslog(LOG_ERR,"2.String to be entered in the file");	
+ 		 return 1;
+	}
+
+	//check if empty string is not passed as filename
+	if(*argv[1]=='\0')
+	{
+		printf("Invalid File name \n");
+		syslog(LOG_ERR,"Invalid File name");
+		return 1;
+	}
+	
+	int fd, wrbytes, closestat;
+
+	//creates a new file or rewrites an existing one
+	fd = creat (argv[1], 0777);
```

Comment:
@DivyeshShashikant suggestion here: Use a const char* to the argument with descriptive name to make your program more readable and easier to modify.

Reply:
Noted. Shall work it next time.